test/src/basic/insert.cpp

```diff
@@ -0,0 +1,506 @@
+/** \file
+ * \brief Tests for ogdf::Graph::insert(...).
+ *
+ * \author Simon D. Fink
+ *
+ * \par License:
+ * This file is part of the Open Graph Drawing Framework (OGDF).
+ *
+ * \par
+ * Copyright (C)<br>
+ * See README.md in the OGDF root directory for details.
+ *
+ * \par
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * Version 2 or 3 as published by the Free Software Foundation;
+ * see the file LICENSE.txt included in the packaging of this file
+ * for details.
+ *
+ * \par
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * \par
+ * You should have received a copy of the GNU General Public
+ * License along with this program; if not, see
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+#include <ogdf/basic/GraphCopy.h>
+#include <ogdf/basic/GraphSets.h>
+#include <ogdf/basic/extended_graph_alg.h>
+#include <ogdf/basic/graph_generators.h>
+#include <ogdf/basic/simple_graph_alg.h>
+
+#include <set>
+
+#include <graphs.h>
+#include <testing.h>
+
+using namespace ogdf;
+using namespace bandit;
+
+void randomTriangulate(Graph& G) {
+	ogdf::makeConnected(G);
+	ogdf::planarEmbed(G);
+
+	CombinatorialEmbedding emb(G);
```

Comment:
I'm not 100% sure why I re-implemented `triangulate` as I would have used that function if it would have fit. It might be that this function generates a random triangulation while the `simple_graph_alg` function generates very regular ones (many subdivision edges to the same vertex)?